pkg/component/runtime/conn_info_server.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package runtime
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	"github.com/elastic/elastic-agent/pkg/core/logger"
+)
+
+const (
+	defaultStopTimeout = 15 * time.Second
+	windows            = "windows"
+)
+
+type connInfoServer struct {
+	log         *logger.Logger
+	listener    net.Listener
+	waitCtx     context.Context
+	stopTimeout time.Duration
+}
+
+func newConnInfoServer(log *logger.Logger, comm Communicator, port int) (*connInfoServer, error) {
```

Comment:
Just reading this in isolation from its use cases its hard to know what problem this is solving.

Finding the definition of `WriteConnInfo` helped a little bit, in that I can see we are serialization a protobuf connection information message and writing it the writer: https://github.com/elastic/elastic-agent/blob/77beb04ffb47fd4b06860844fbdb7ef803276ca8/pkg/component/runtime/runtime_comm.go#L112-L119

It's not obvious who should be dialing us on localhost to obtain this information, or why.

I assume this is a special method of boostrapping the control protocol connection for components that run as independent services? Can we document this somewhere?

Could you create need a service runtime README.md file or package level documentation comment explaining how this works, and how it is all wired together? I think that would help people unfamiliar with this part of the agent ramp up on it faster.

Reply:
It was not documented in V1, so had to grep around to figure out how it works.
Sure can create a brief readme at the package level.

Reply:
README is great, thank you!